### ISSUE TITLE
feat: jsr publish should exclude files by default

### ISSUE THREAD

**Issue by vikingair** [jsr-io/jsr#566]
I think it is wrong to publish the lock file along with the package.

I used `pnpm` and `pnpm dlx jsr publish` + `pnpm dlx jsr add ...` to see that my `pnpm-lock.yaml` was published.

Is there any good reason to do that?

I could work around that issue by specifying "include" files manually, but shouldn't this be avoided by default?

**Comment by fdc-viktor-luft** [comment]
Yes, if it is the same reason, it is just fine to close them all :+1:

But as an counter argument: My "node_modules" and any files and directories starting with "." weren't published.

I think JSR could use better defaults in case "include" wasn't explicitly specified or alternatively make it mandatory to specify the "include".

**Comment by lucacasonato** [comment]
Note that you do not need to use `include`. You can just `exclude` the specific lockfile, or add the lockfile to your `.gitignore`.

**Comment by fdc-viktor-luft** [comment]
To the `.gitignore`: That would be kind of unintuitive and in case of the lock file not even recommended practice.

As for the `exclude`: Doesn't make a real difference to me to use this or `include`. I think JSR (also trying to address new unexperienced authors) should provide good defaults and allow to specify `include` and `exclude` if those are not well suited enough.

**Comment by lucacasonato** [comment]
I think there is not much we can do here - requiring the `include` field to always be set would make publishing much more cumbersome.
